src/sql-parser/src/parser.rs

```diff
@@ -1960,7 +1960,26 @@ impl<'a> Parser<'a> {
             let debezium_mode = DbzMode::Plain;
             Envelope::Debezium(debezium_mode)
         } else if self.parse_keyword(UPSERT) {
-            Envelope::Upsert
+            let order_by = if self.consume_token(&Token::LParen) {
+                self.expect_keywords(&[ORDER, BY])?;
+                self.expect_token(&Token::LParen)?;
+
+                let order_by_cols = self.parse_comma_separated(|parser| {
+                    let order_col = parser.parse_identifier()?;
+                    // only optional ASC allowed
+                    let _order = parser.parse_keyword(ASC);
+
+                    Ok(order_col)
+                })?;
+
+                self.expect_token(&Token::RParen)?;
+                self.expect_token(&Token::RParen)?;
+                order_by_cols
+            } else {
+                vec![]
+            };
+
+            Envelope::Upsert { order_by }
```

Comment:
Can you add some parser tests here: https://github.com/MaterializeInc/materialize/blob/main/src/sql-parser/tests/testdata/ddl

Reply:
Will do 👍

Reply:
Added!